pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -243,6 +258,33 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 	c.serviceLister = kubeClient.KubeInformer().Core().V1().Services().Lister()
 	registerHandlers(c.serviceInformer, c.queue, "Services", c.onServiceEvent, nil)
 
+	if enableEventing.Get() {
+		// To get events posted by other Istiod instances, in order to sync and support debug tools.
+		// This will also allow us to get readiness/status when VM health check is done, and possibly get faster
```

Comment:
why is:
istiod -> k8s api-server (events) -> other istiod (k8s api-server watch)

and faster than:
istiod -> k8s api-server (workload entry) -> other istiod (k8s api-server watch)

?

Is events optimized in any way for this use case?

Reply:
Not sure what does it have to do with WorkloadEntry - this is about Pods. 
We can also associate the events with WorkloadEntry - but some of the events will be 'global'. 

AFAIK Events are optimized for this use case - they can be collapsed and I think the're not persisted to etcd ( memory only).
We can also send events even if the pod is not yet visible ( for example envoy connects before we get the Pod object), or if the pod is dead ( disconnect events - pod is gone ). Obviously 'kubectl describe' will not show anything, but kubectl get events and other istiods will see the disconnect.

Reply:
VM healthcheck will end up in workload entry, the comment suggest we use it for this purpose. My main point here is I am very skeptical of using events for endpoint status

Reply:
See above. The events will be used for Istiod to discover each other - across clusters. It's not perfect - if I run a Istiod on a VM without a public IP, it may not be reached. 

The real problem is how to build the mesh of connections between istiod instances - the 'bus'. For single cluster is easy, just use the Endpoints. 

In any case - the first goal is debug info. Crash looping istiods will be easier to find for example.